vumi/components/message_store.py

```diff
@@ -23,6 +24,29 @@
     EventMigrator, InboundMessageMigrator, OutboundMessageMigrator)
 
 
+def to_reverse_timestamp(vumi_timestamp):
+    """
+    Turn a vumi_date-formatted string into a string that sorts in reverse order
+    and can be turned back into a timestamp later.
+
+    This is done by converting to a unix timestamp and subtracting it from
+    0xffffffffff (2**40 - 1) to get a number well outside the range
+    representable by the datetime module. The result is returned as a
+    hexadecimal string.
+    """
+    timestamp = timegm(parse_vumi_date(vumi_timestamp).timetuple())
+    return "%X" % (0xffffffffff - timestamp)
```

Comment:
Is there a reason why we're not using `2^^31 - 1` or `2^^63 - 1` (for 32 and 64 signed ints)? Although 32 bits will run out relatively soon.